cmd/integration_test.go

```diff
@@ -1534,3 +1534,122 @@ func TestPrometheusRemoteWriteOutput(t *testing.T) {
 
 	assert.Contains(t, stdOut, "output: Prometheus remote write")
 }
+
+func TestTracingClient(t *testing.T) {
+	t.Parallel()
+	tb := httpmultibin.NewHTTPMultiBin(t)
+
+	gotRequests := 0
+
+	tb.Mux.HandleFunc("/tracing", func(w http.ResponseWriter, r *http.Request) {
+		gotRequests++
+		assert.NotEmpty(t, r.Header.Get("traceparent"))
+		assert.Len(t, r.Header.Get("traceparent"), 55)
+	})
+
+	script := tb.Replacer.Replace(`
+		import http from "k6/http";
+		import { check } from "k6";
+		import tracing from "k6/experimental/tracing";
+
+		const instrumentedHTTP = new tracing.Client({
+			propagator: "w3c",
+		})
+
+		export default function () {
+			instrumentedHTTP.del("HTTPBIN_IP_URL/tracing");
+			instrumentedHTTP.get("HTTPBIN_IP_URL/tracing");
+			instrumentedHTTP.head("HTTPBIN_IP_URL/tracing");
+			instrumentedHTTP.options("HTTPBIN_IP_URL/tracing");
+			instrumentedHTTP.patch("HTTPBIN_IP_URL/tracing");
+			instrumentedHTTP.post("HTTPBIN_IP_URL/tracing");
+			instrumentedHTTP.put("HTTPBIN_IP_URL/tracing");
+			instrumentedHTTP.request("GET", "HTTPBIN_IP_URL/tracing");
+		};
+	`)
+
+	ts := getSingleFileTestState(t, script, []string{"--out", "json=results.json"}, 0)
+	newRootCommand(ts.globalState).execute()
+
+	assert.Equal(t, 8, gotRequests)
+
+	jsonResults, err := afero.ReadFile(ts.fs, "results.json")
+	require.NoError(t, err)
+
+	for _, jsonLine := range bytes.Split(jsonResults, []byte("\n")) {
+		if len(jsonLine) == 0 {
+			continue
+		}
+
+		var line sampleEnvelope
+		require.NoError(t, json.Unmarshal(jsonLine, &line))
+
+		if line.Type != "Point" {
+			continue
+		}
+
+		// Filter metric samples which are not related to http
+		if !strings.HasPrefix(line.Metric, "http_") {
+			continue
+		}
```

Comment:
because of those continues (and hte above one) there is a chance that a test will just never go to the below lines. 

I would recommend a boolean that you set to see that at least once you got through the whole check

Reply:
Good point, I modified the test to do just that indeed 👍🏻